### PR TITLE
feat: added an ability to save the file right after creating

### PR DESCRIPTION
**Pull request by Toxer10**
## Changes

- added an ability to save the file right after creating

### CODE REVIEW

**Comment by olensmar** [comment]
thanks @EduardIvanov22 - i suggest the following changes:
- change the label to the checkbox to  "Save to folder {currentFolder}" - and automatically save to that folder (no prompting)
- currentFolder defaults to the current root folder
- add a "Change folder" button (after the label?) that the user can select to change the folder (which would open the dialog you already have) - the label would change accordingly once selected
- remember the last selected folder (if the user changes it from the default root folder)

what do you think? @devcatalin ? @erdkse ?

**Comment by devcatalin** [comment]
I agree with all the points @olensmar mentioned.

For implementation I would just use the `createUnsavedResource` and then `saveUnsavedResource` giving it the path of the selected folder

**Comment by olensmar** [comment]
thanks @devcatalin - @EduardIvanov22 you'll need to merge main into your branch - i recently made some changes to the dialogs and saveUnsavedResource action

**Comment by Toxer10** [comment]
image
What do you think of this approach? @olensmar @devcatalin

**Comment by olensmar** [comment]
looks good - the drop-down would contain all folders under the current root folder - correct? (with their relative path to the root folder)

File: src/components/organisms/NewResourceWizard/NewResourceWizard.tsx [hunk 1]
```diff
@@ -103,26 +115,74 @@ const NewResourceWizard = () => {
...
+        apiVersion: formValues.apiVersion,
       },
       dispatch,
       jsonTemplate
     );
 
+    const fullFileName = getFullFileName(formValues.name);
+
+    if (shouldSaveToFolder) {
+      setTimeout(() => {
```
Comment (olensmar) [inline src/components/organisms/NewResourceWizard/NewResourceWizard.tsx:154]: could you simply add an optional savePath property to the createUnsavedResource method above - which would save the resource after creating it? (if the path was provided)

File: src/components/organisms/NewResourceWizard/NewResourceWizard.tsx [hunk 2]
```diff
@@ -103,26 +115,74 @@ const NewResourceWizard = () => {
...
+      Object.entries(fileMap)
+        .map(([key, value]) => ({folderName: key, isFolder: Boolean(value.children)}))
+        .filter(file => file.isFolder),
+    [fileMap]
+  );
+
+  const renderSelectOptions = () => {
+    return foldersList.map(folder => (
+      <Option key={folder.folderName} value={folder.folderName}>
+        {folder.folderName}
```
Comment (olensmar) [inline src/components/organisms/NewResourceWizard/NewResourceWizard.tsx:181]: suggest to remove the leading '/'

File: src/components/organisms/NewResourceWizard/NewResourceWizard.tsx (hunk 1 above)
Comment (Toxer10) [reply src/components/organisms/NewResourceWizard/NewResourceWizard.tsx:154]: Sure

File: src/components/organisms/NewResourceWizard/NewResourceWizard.tsx (hunk 2 above)
Comment (Toxer10) [reply src/components/organisms/NewResourceWizard/NewResourceWizard.tsx:181]: Sure

Comment (Toxer10) [reply src/components/organisms/NewResourceWizard/NewResourceWizard.tsx:181]: Done

**Review by olensmar — APPROVED** [review]

**Pull request merged**